### DISCUSSION TITLE
I have a strange issue in my production website

### DISCUSSION THREAD

**Discussion by vnetonline** [oqtane/oqtane.framework#3189] [Q&A]
@sbwalker & @leigh-pointer When I click on any users profile I get the following errorr

![image](https://github.com/oqtane/oqtane.framework/assets/15242923/e39ad42c-f734-48ce-80c4-ffb4d19befd0)

When I look at the Event Log i see 

![image](https://github.com/oqtane/oqtane.framework/assets/15242923/27834aac-1446-446b-8e60-b3ede62342c9)

and also this 

![image](https://github.com/oqtane/oqtane.framework/assets/15242923/d4333405-d032-4250-bb99-a64520a4c6ef)

I can't figure out why ... can any one help ?

I am running 4.2 and when I run it on my local machine it works fine!

**Comment by sbwalker** [comment]
@vnetonline what hosting environment are you running your production site on? One item I am noticing is that the Url in the error message contains %5C encoded characters - which are forward slashes ("\\"). However the Path field in the Folder table is stored using back slashes ("/") - which would be encoded as %2F.

**Comment by sbwalker** [comment]
This is interesting...

The User.cs model contains:

```
        public string FolderPath
        {
            get => "Users\\" + UserId.ToString() + "\\";
        }
```

And the FolderService contains:

```
        public async Task<Folder> GetFolderAsync(int siteId, [NotNull] string folderPath)
        {
            var path = WebUtility.UrlEncode(folderPath);
            return await GetJsonAsync<Folder>($"{ApiUrl}/{siteId}/{path}");
        }
```

and the FolderController contains:

```
        [HttpGet("{siteId}/{path}")]
        public Folder GetByPath(int siteId, string path)
        {
            var folderPath = WebUtility.UrlDecode(path).Replace("\\", "/");
            folderPath = (folderPath == "/") ? "" : folderPath;
            if (!folderPath.EndsWith("/") && folderPath != "")
            {
                folderPath += "/";
            }
            Folder folder = _folders.GetFolder(siteId, folderPath);
            if (folder != null && folder.SiteId == _alias.SiteId && _userPermissions.IsAuthorized(User, PermissionNames.View, folder.PermissionList))
            {
                return folder;
            }
            else
            {
                if (folder != null)
                {
                    _logger.Log(LogLevel.Error, this, LogFunction.Security, "Unauthorized Folder Get Attempt {Path} For Site {SiteId}", path, siteId);
                    HttpContext.Response.StatusCode = (int)HttpStatusCode.Forbidden;
                }
                else
                {
                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
                }
                return null;
            }
        }
```

In earlier versions of Oqtane, folder paths were stored using forward slashes {"\\") however this was later changed to better support non-Windows environments - which is why there is logic above to convert ("\\") to ("/"). However it does not explain your production issue.

I have seen some web servers behave strangely in the past when it comes to path delimiters - some web servers think that %5C characters are malicious attempts to do path traversal and they filter them.

**Comment by sbwalker** [comment]
@vnetonline I just remembered that we ran into this problem before... and it is caused by inconsistent behavior by web servers when interpreting Url paths. This problem was raised previously in regards to alias resolution... and the solution was to pass the encoded path parameter in the querystring rather than as part of the url path.

**Comment by vnetonline** [comment]
Hmmm I am running on Azure web service and it was working all this time I haven't changed any setting

**Reply by sbwalker** [reply]
@vnetonline so you are saying that the user profile was working fine previously in production? I thought you said that it was working fine in your development environment but not production.

Note that inconsistent behavior I described above is only related to API calls which use encoded slash characters in the Url path - which is a very rare occurrence (I don't think there are any other cases in the framework).

**Reply by vnetonline** [reply]
It was working in both development and production. But now it is not working in production all of the sudden after I upgraded to 4.2 

I haven't changed anything I have used the deploy button which was the first thing I worked on 3 to 4 months ago

**Reply by vnetonline** [reply]
It seems I am cursed .... lol always finding curly ones which no one else faces

**Reply by sbwalker** [reply]
I do not see anything in the Folder API code which has changed recently which could have caused this in your production environment. One recent change was made in the ServiceBase to catch errors like this and log them so that it is easier to investigate (previously you would only get the generic "Value cannot be Null. Parameter name 'source'" which was difficult to diagnose).

**Comment by vnetonline** [comment]
@sbwalker any ideas what I should do to resolve this ??

**Comment by sbwalker** [comment]
I will make a change to the Folder API to use a Querystring parameter rather than a Url segment.

**Comment by sbwalker** [comment]
I am able to repro this on Azure as well now... I wonder if Azure made a recent configuration change related to Url handling.

**Reply by vnetonline** [reply]
Good to know I am not crazy

**Comment by sbwalker** [comment]
@vnetonline see #3190

**Comment by vnetonline** [comment]
I confirm this Fix works, I had the same solution but didn't submit the pull request because I didn't want to break anything and I thought it may have been specific to my deployment environment.

Are you using Pipelines (DevOps) to deploy your websites @sbwalker ??

**Comment by sbwalker** [comment]
@vnetonline I will release a 4.0.3 early next week which includes this fix

**Reply by vnetonline** [reply]
Thanks @sbwalker much appreciated
